### PR TITLE
Add getRetiringFederatorBtcPublicKey tests

### PR DESCRIPTION
**Pull request by julia-zack**
- Add `getRetiringFederatorBtcPublicKey` tests
- Rename `getRetiringFederatorPublicKey` to `getRetiringFederatorBtcPublicKey` in `FederationSupport` and in `BridgeSupport` to be more accurate with what the method does

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 1]
```diff
@@ -1189,6 +1203,92 @@ void getRetiringFederationCreationBlockNumber_withNewFederationActive_returnsOld
...
+                .build();
+
+            byte[] retiringFederatorBtcPublicKey = federationSupport.getRetiringFederatorBtcPublicKey(0);
+            assertThat(retiringFederatorBtcPublicKey, is(oldFederation.getBtcPublicKeys().get(0).getPubKey()));
+        }
+
+        @ParameterizedTest
+        @Tag("getRetiringFederatorBtcPublicKey")
+        @MethodSource("newFederationActiveActivationArgs")
+        void getRetiringFederatorBtcPublicKey_withNegativeIndex_throwsIndexOutOfBoundsException(
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:1251]: this test case needs an existing retiring federation

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 2]
```diff
@@ -1189,6 +1203,92 @@ void getRetiringFederationCreationBlockNumber_withNewFederationActive_returnsOld
...
+                .withActivations(activations)
+                .build();
+
+            assertThrows(IndexOutOfBoundsException.class, () -> federationSupport.getRetiringFederatorBtcPublicKey(-1));
+        }
+
+        @ParameterizedTest
+        @Tag("getRetiringFederatorBtcPublicKey")
+        @MethodSource("newFederationActiveActivationArgs")
+        void getRetiringFederatorBtcPublicKey_withIndexGreaterThanRetiringFederationSize_throwsIndexOutOfBoundsException(
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:1271]: same as above

**Review by wilmerrootstock — APPROVED** [review]
Good Job, @julia-zack!
LGTM!

**Review by marcos-iov — APPROVED** [review]

**Review by apancorb — APPROVED** [review]
LGTM

**Review by nathanieliov — APPROVED** [review]

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**